packages/flutter/test/widgets/two_dimensional_viewport_test.dart

```diff
@@ -0,0 +1,455 @@
+    required super.verticalAxisDirection,
+    required super.horizontalOffset,
+    required super.horizontalAxisDirection,
+    required super.delegate,
+    required super.mainAxis,
+    super.cacheExtent,
+    super.clipBehavior = Clip.hardEdge,
+  });
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return RenderSimpleTableViewport(
+      horizontalOffset: horizontalOffset,
+      horizontalAxisDirection: horizontalAxisDirection,
+      verticalOffset: verticalOffset,
+      verticalAxisDirection: verticalAxisDirection,
+      mainAxis: mainAxis,
+      delegate: delegate,
+      childManager: context as RenderObjectElement,
+      cacheExtent: cacheExtent,
+      clipBehavior: clipBehavior,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, RenderSimpleTableViewport renderObject) {
+    renderObject
+      ..horizontalOffset = horizontalOffset
+      ..horizontalAxisDirection = horizontalAxisDirection
+      ..verticalOffset = verticalOffset
+      ..verticalAxisDirection = verticalAxisDirection
+      ..mainAxis = mainAxis
+      ..delegate = delegate
+      ..cacheExtent = cacheExtent
+      ..clipBehavior = clipBehavior;
+  }
+}
+
+class RenderSimpleTableViewport extends RenderTwoDimensionalViewport {
+  RenderSimpleTableViewport({
+    required super.horizontalOffset,
+    required super.horizontalAxisDirection,
+    required super.verticalOffset,
+    required super.verticalAxisDirection,
+    required super.delegate,
+    required super.mainAxis,
+    required super.childManager,
+    super.cacheExtent,
+    super.clipBehavior = Clip.hardEdge,
+    this.applyDimensions = true,
+    this.useParentSize = true,
+    this.setLayoutOffset = true,
+  });
+
+  final bool applyDimensions; // Testing error message
+  final bool useParentSize; // Testing error message
+  final bool setLayoutOffset; // Testing error message
+
+  @override
+  void layoutChildSequence() {
+    // Really simple table implementation for testing.
+    // Every child is 200x200 square
+    final double horizontalPixels = horizontalOffset.pixels;
+    final double verticalPixels = verticalOffset.pixels;
+    final int leadingColumn = math.max((horizontalPixels / 200).floor(), 0);
+    final int leadingRow = math.max((verticalPixels / 200).floor(), 0);
+    final int trailingColumn = math.min(((horizontalPixels + viewportDimension.width) / 200).ceil(), 99);
+    final int trailingRow = math.min(((verticalPixels + viewportDimension.height) / 200).ceil(), 99);
+
+    double xLayoutOffset = (leadingColumn * 200) - horizontalOffset.pixels;
+    for(int column = leadingColumn; column <= trailingColumn; column++) {
+      double yLayoutOffset =  (leadingRow * 200) - verticalOffset.pixels;
+      for (int row = leadingRow; row <= trailingRow; row++) {
+        final RenderBox child = buildOrObtainChildFor(ChildVicinity(xIndex: row, yIndex: column))!;
+        child.layout(constraints, parentUsesSize: useParentSize);
```

Comment:
shouldn't the child be laid out with constraints that force it to be 200x200 based on the comment at the beginning of this method?